libexec/bats-core/bats

```diff
@@ -84,94 +84,111 @@ expand_path() {
   printf -v "$result" '%s/%s' "$dirname" "${path##*/}"
 }
 
-# argument: exclude_literal
+# argument: file
 is_bats_file() {
-  [[ "$1" == *".bats"* ]]
+  [[ "$1" == *"${BATS_FILE_EXTENSION:-.bats}"* ]];
 }
 
-# argument suite_filename
-is_excluded() {
-  if [[ -z "$exclude_files" ]]; then
-    return 1
-  fi
-  
-  local suite_filename="$1"
-  read -ra EXCLUDE_FILES <<< "$exclude_files"
+# arguments:
+#  filtered_suite_filenames ('out' param)
+#  exclude_files 
+#  suite_filenames
+filter_suite() {
```

Comment:
**naming:**
implemented the filter function of suite filenames more explicit.

**logic:**
almost the same as before except that EXCLUDE_FILES only need to be created once.
=> looping in line 480 moved one level up into this function (sorry cannot create a permalink to the line)